fluence-faas/src/faas.rs

```diff
@@ -80,8 +81,34 @@ impl FluenceFaaS {
         })
     }
 
+    /// Searches for modules in `config.core_modules_dir`, loads only those in the `names` set
+    pub fn with_module_names<C>(names: &mut HashSet<String>, config: C) -> Result<Self>
```

Comment:
This signature looks a little bit discouraging, because from the function name I don't have any idea why names should be passed by mut ref. What about using searching instead of removing in a closure? And to check that all modules's been found we could just compare modules and names lens. In this case we can pass names by const ref.